### PR TITLE
Jupyter ISTIO routing rule - Ksonnet

### LINKED ISSUE #2964: ISTIO routing rule for jupyter web app
We will need an ISTIO routing rule for the Jupyter web app to replace Ambassador

### PR DESCRIPTION
**Pull request by gabrielwen**
Relates to #2964 

Relates to #2978

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/3028)

### CODE REVIEW

**Comment by gabrielwen** [comment]
Verified it's working with manual deployment.

File: kubeflow/jupyter/jupyter-web-app.libsonnet [hunk 1]
```diff
@@ -216,6 +216,58 @@
...
+        ],
+        gateways: [
+          "kubeflow-gateway",
+        ],
+        http: [
+          {
+            match: [
+              {
+                uri: {
+                  prefix: "/" + params.prefix + "/",
```
Comment (lluunn) [inline kubeflow/jupyter/jupyter-web-app.libsonnet]: do we want exact or prefix?
also, do we need the trailing /?

File: kubeflow/jupyter/prototypes/jupyter-web-app.jsonnet [hunk 2]
```diff
@@ -8,6 +8,8 @@
 // @optionalParam policy string Always imagePullPolicy for the UI's image
 // @optionalParam prefix string jupyter The prefix under which the app is accessed
 // @optionalParam rokSecretName string secret-rok-{username} The name of the secret containing user's credentials for Arrikto Rok
+// @optionalParam injectIstio string false Whether to inject istio sidecar; should be true or false.
+// @optionalParam clusterDomain string svc.cluster.local DNS config to cluster domain.
```
Comment (lluunn) [inline kubeflow/jupyter/prototypes/jupyter-web-app.jsonnet]: do we have to do this clusterDomain for all components?

File: kubeflow/jupyter/jupyter-web-app.libsonnet (hunk 1 above)
Comment (gabrielwen) [reply kubeflow/jupyter/jupyter-web-app.libsonnet]: I tried to keep the same semantics - in ambassador config it's prefix, and it has trailing /

File: kubeflow/jupyter/prototypes/jupyter-web-app.jsonnet (hunk 2 above)
Comment (gabrielwen) [reply kubeflow/jupyter/prototypes/jupyter-web-app.jsonnet]: yap, and I did add it to all components.

**Pull request merged**